### PR TITLE
IDE-1049 Duplicate entry in clienttools pulldown for compiler

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Signed-off-by: David de Hilster <dehilsterlexis>

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

File: comms/EclCC.cpp [hunk 1]
```diff
@@ -725,11 +725,26 @@ boost::filesystem::path EclCCSubPath(const boost::filesystem::path & directory)
     return directory / "clienttools" / "bin" / "eclcc.exe";
 }
 
-unsigned int FindAllCEclCC(const boost::filesystem::path & progFiles, CEclCCVector & results)
+void PushUniqueEclCC(CEclCCVector& results, boost::filesystem::path eclccPath)
```
Comment (GordonSmith) [inline comms/EclCC.cpp:728]: Should be `const boost::filesystem::path &eclccPath`

File: comms/EclCC.cpp [hunk 2]
```diff
@@ -725,11 +725,26 @@ boost::filesystem::path EclCCSubPath(const boost::filesystem::path & directory)
...
 {
-    if (boost::filesystem::exists(progFiles)) {
+    bool found = false;
+    for (auto& c : results)
```
Comment (GordonSmith) [inline comms/EclCC.cpp:731]: Should be using `std::find`

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: comms/EclCC.cpp (hunk 1 above)
Comment (dehilsterlexis) [reply comms/EclCC.cpp]: Done.

File: comms/EclCC.cpp (hunk 2 above)
Comment (dehilsterlexis) [reply comms/EclCC.cpp]: I used "find_if" given it is a comparison of a value in the object.

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

File: comms/EclCC.cpp [hunk 3]
```diff
@@ -725,11 +725,19 @@ boost::filesystem::path EclCCSubPath(const boost::filesystem::path & directory)
     return directory / "clienttools" / "bin" / "eclcc.exe";
 }
 
-unsigned int FindAllCEclCC(const boost::filesystem::path & progFiles, CEclCCVector & results)
+void PushUniqueEclCC(CEclCCVector& results, const boost::filesystem::path eclccPath)
```
Comment (GordonSmith) [inline comms/EclCC.cpp:728]: Should be `const boost::filesystem::path &eclccPath`

File: comms/EclCC.cpp [hunk 4]
```diff
@@ -725,11 +725,19 @@ boost::filesystem::path EclCCSubPath(const boost::filesystem::path & directory)
...
 {
-    if (boost::filesystem::exists(progFiles)) {
+    bool found = false;
```
Comment (GordonSmith) [inline comms/EclCC.cpp:730]: Is this used?

File: comms/EclCC.cpp [hunk 5]
```diff
@@ -725,11 +725,19 @@ boost::filesystem::path EclCCSubPath(const boost::filesystem::path & directory)
...
+    auto it = std::find_if(results.begin(), results.end(), [&eclccPath](const CEclCCAdapt& obj) {return (_tcsicmp(obj.get()->GetCompilerFilePath().c_str(), eclccPath.c_str()) == 0); });
```
Comment (GordonSmith) [inline comms/EclCC.cpp:731]: `std::any_of` returns a bool which may be preferable, but checking against `results.end()` is equally ok.

File: comms/EclCC.cpp [hunk 6]
```diff
@@ -725,11 +725,19 @@ boost::filesystem::path EclCCSubPath(const boost::filesystem::path & directory)
...
+    if (results.size() == 0 || it == results.end()) {
```
Comment (GordonSmith) [inline comms/EclCC.cpp:732]: Why test for `results.size() == 0` here?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

**Comment by dehilsterlexis** [comment]
@GordonSmith ping

File: comms/EclCC.cpp (hunk 3 above)
Comment (GordonSmith) [reply comms/EclCC.cpp:728]: @dehilsterlexis bump

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

**Pull request merged**